### ISSUE TITLE
iOS

### ISSUE THREAD

**Issue by AKZHW** [SDWebImage/SDWebImage#3450]
### New Issue Checklist

* [ ] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [ ] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [ ] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | e.g. ios / macos / tvos / watchos
 Platform Version        | e.g. 12.0 / 10.14.0 / 12.0 / 5.0
 SDWebImage Version      | e.g. 5.0.0 / 4.4.0
 Integration Method      | e.g. carthage / cocoapods / manually
 Xcode Version           | e.g. Xcode 11 / Xcode 10
 Repro rate              | e.g. all the time (100%) / sometimes x% / only once
 Repro with our demo prj | e.g. does it happen with our demo project?
 Demo project link       | e.g. link to a demo project that highlights the issue

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

**Comment by AKZHW** [comment]
iOS 16+ , app enterbackground  Crash, crash log


0 | CoreServicesInternal | _FileCacheLock(__FileCache const*) | - | -
-- | -- | -- | -- | --
1 | CoreServicesInternal | _FileCacheGetFlags | - | -
2 | CoreServicesInternal | _FileCacheGetFlags | - | -
3 | CoreServicesInternal | _GetDirectoryURLs(_CFURLEnumerator*) | - | -
4 | CoreServicesInternal | _URLEnumeratorGetNextURL | - | -
5 | Foundation | -[NSURLDirectoryEnumerator nextObject] | - | -
6 | CoreFoundation | -[NSEnumerator countByEnumeratingWithState:objects:count:] | - | -
7 | LJShell | -[SDDiskCache removeExpiredData] | SDDiskCache.m | 199
8 | LJShell | __50-[SDImageCache deleteOldFilesWithCompletionBlock:]_block_invoke | SDImageCache.m | 662
9 | libdispatch.dylib | _dispatch_call_block_and_release | - | -
10 | libdispatch.dylib | _dispatch_client_callout | - | -
11 | libdispatch.dylib | _dispatch_lane_serial_drain | - | -
12 | libdispatch.dylib | _dispatch_lane_invoke | - | -
13 | libdispatch.dylib | _dispatch_workloop_worker_thread | - | -
14 | libsystem_pthread.dylib | _pthread_wqthread

**Comment by dreampiggy** [comment]
I don't think this issue is SDWebImage related. Maybe Apple's bug and need firing radar:

https://feedbackassistant.apple.com/

This use the standard Foundation API (which documentations says no lock is needed)

**Comment by dreampiggy** [comment]
See the similiar thing in 

https://github.com/onevcat/Kingfisher/issues/1993
https://github.com/onevcat/Kingfisher/issues/2037

**Comment by T-Damer** [comment]
Looks like I have something similar too
